baseplate/secrets/fetcher.py

```diff
@@ -205,23 +213,27 @@ def _vault_aws_auth(self):
         """
         identity_document = fetch_instance_identity()
         nonce = load_nonce()
+        if nonce is None:
```

Comment:
`if not nonce:` unless you think nonce could legitimately be falsey but non-None